crates/schema/src/lib.rs

```diff
@@ -25,19 +25,12 @@ use crate::component::ComponentSchemas;
 use crate::deployment::DeploymentSchemas;
 use restate_types::{Version, Versioned};
 
-/// Immutable view of the schema registry.
+/// Schema information which automatically loads the latest version when accessing it.
 ///
 /// Temporary bridge until users are migrated to directly using the metadata
-/// provided schema registry.
-#[derive(Debug, Default, Clone)]
-pub struct SchemaView(Arc<ArcSwap<SchemaInformation>>);
-
-impl SchemaView {
-    /// Update current view with new schema registry.
-    pub fn update(&self, schema_registry: Arc<SchemaInformation>) {
-        self.0.store(schema_registry);
-    }
-}
+/// provided schema information.
+#[derive(Debug, Default, Clone, derive_more::From)]
+pub struct UpdatingSchemaInformation(Arc<ArcSwap<SchemaInformation>>);
```

Comment:
A mildly confusing name? Maybe `UpdateableSchema` like `UpdateableConfiguration`?

Also, what was the intention behind adding `Information` as suffix?

Reply:
Yeah, I am also not super happy with the name. The reason why I chose a different name from `Updateable*` is because unlike `UpdateableConfiguration` and other `Updateables` there is no explicit `load` command to fetch the latest version. It kind of happens under the hood as part of the different `*Resolver` implementations. I can also change it to `UpdateableSchema` if this makes it less confusing.

+1 for removing `*Information`.

Reply:
Ah, I see your point, I'd be okay if it's `UpdateableSchema` that doesn't have a `load()` function.

Reply:
Ok, then I'll change it.